### PR TITLE
fix for #2147

### PR DESCRIPTION
**Pull request by lachlan-00**

### CODE REVIEW

File: lib/class/xml_data.class.php [hunk 1]
```diff
@@ -489,9 +489,12 @@ public static function tags($tags)
      */
     public static function artists($artists, $include = [], $full_xml = true, $user_id = false)
     {
-        if (null == $include) {
+        if ($include == null || $include == '') {
             $include = array();
         }
+        if (is_string($include)) {
```
Comment (AshotN) [inline lib/class/xml_data.class.php:495]: `$include` is expected to be of type `array` as per the docs. Why hare we allowing someone to send in a `string`?

Comment (lachlan-00) [reply lib/class/xml_data.class.php:495]: i'll check where this is called from, back here was just breakfixing so it's probably simple to clean up

Comment (lachlan-00) [reply lib/class/xml_data.class.php:495]: explode on the input?
```
echo XML_Data::artists($artists, $input['include'], $user->id);
```

Comment (AshotN) [reply lib/class/xml_data.class.php:495]: I think that would make more sense, would get rid of a check. Reduce complexity.

Comment (lachlan-00) [reply lib/class/xml_data.class.php:495]: yeah, now that i've documented all these functions they need to actually send what we want.
doing it in #2347 as well

**Pull request merged**